.Net 9.0 NSwagExe_Net90 failing to generate C# client

**Describe the bug**

Hello! Thank you for the excellent project!

Since upgrading my test project (previously working for .Net 7 and .Net 8) to .Net 9 I'm now getting the following error when generating my C# client via post-build step:

```
  <Target Name="NSwag" AfterTargets="Build">
    <Exec Command="$(NSwagExe_Net90) run nswag.json /variables:Configuration=$(Configuration)" />
  </Target>
```

```
  CNR.Example.API net9.0 failed with 1 error(s) (7.6s) → CNR.Example.API\bin\Debug\net9.0\CNR.Example.API.dll
    NSwag command line tool for .NET Core Net90, toolchain v14.2.0.0 (NJsonSchema v11.1.0.0 (Newtonsoft.Json v13.0.0.0))
    Visit http://NSwag.org for more information.
    NSwag bin directory: C:\Users\Christopher.Cameron\.nuget\packages\nswag.msbuild\14.2.0\tools\Net90

    Executing file 'nswag.json' with variables 'Configuration=Debug'...
    System.ArgumentException: The path is empty. (Parameter 'path')
       at System.IO.Path.GetFullPath(String path)
       at NSwag.Commands.Generation.AspNetCore.ProjectMetadata.GetProjectMetadata(String file, String buildExtensionsDir, String framework, String configuration, String runtime, Boolean noBuild, String outputPath, IConsoleHost console) in /_/src/NSwag.Commands/Commands/Generation/AspNetCore/ProjectMetadata.cs:line 113
       at NSwag.Commands.Generation.AspNetCore.AspNetCoreToOpenApiCommand.RunAsync(CommandLineProcessor processor, IConsoleHost host) in /_/src/NSwag.Commands/Commands/Generation/AspNetCore/AspNetCoreToOpenApiCommand.cs:line 70
       at NSwag.Commands.NSwagDocumentBase.GenerateSwaggerDocumentAsync() in /_/src/NSwag.Commands/NSwagDocumentBase.cs:line 270
       at NSwag.Commands.NSwagDocument.ExecuteAsync() in /_/src/NSwag.Commands/NSwagDocument.cs:line 67
       at NSwag.Commands.Document.ExecuteDocumentCommand.ExecuteDocumentAsync(IConsoleHost host, String filePath) in /_/src/NSwag.Commands/Commands/Document/ExecuteDocumentCommand.cs:line 76
       at NSwag.Commands.Document.ExecuteDocumentCommand.RunAsync(CommandLineProcessor processor, IConsoleHost host) in /_/src/NSwag.Commands/Commands/Document/ExecuteDocumentCommand.cs:line 33
       at NConsole.CommandLineProcessor.ProcessSingleAsync(String[] args, Object input)
       at NConsole.CommandLineProcessor.ProcessAsync(String[] args, Object input)
       at NSwag.Commands.NSwagCommandProcessor.ProcessAsync(String[] args) in /_/src/NSwag.Commands/NSwagCommandProcessor.cs:line 65
    C:\Workspace\cnr-common-asp-2\src\CNR.Example.API\CNR.Example.API.csproj(36,5): error MSB3073: The command "dotnet "C:\Users\Christopher.Cameron\.nuget\packages\nswag.msbuild\14.2.0\buildTransitive\../tools/Net90/dotnet-nswag.dll" run nswag.json /variables:Configuration=Debug" exited with code -1.

Build failed with 1 error(s) and 8 warning(s) in 10.7s
```

**Version of NSwag toolchain, computer and .NET runtime used**

`NSwag command line tool for .NET Core Net90, toolchain v14.2.0.0 (NJsonSchema v11.1.0.0 (Newtonsoft.Json v13.0.0.0))`

**To Reproduce**

Here's my nswag.json configuration:

```
{
  "runtime": "Net90",
  "defaultVariables": null,
  "documentGenerator": {
    "aspNetCoreToOpenApi": {
      "project": "CNR.Example.API.csproj",
      "msBuildProjectExtensionsPath": null,
      "configuration": "$(Configuration)",
      "runtime": null,
      "targetFramework": null,
      "noBuild": true,
      "msBuildOutputPath": null,
      "verbose": true,
      "workingDirectory": null,
      "requireParametersWithoutDefault": false,
      "apiGroupNames": null,
      "defaultPropertyNameHandling": "Default",
      "defaultReferenceTypeNullHandling": "Null",
      "defaultDictionaryValueReferenceTypeNullHandling": "NotNull",
      "defaultResponseReferenceTypeNullHandling": "NotNull",
      "generateOriginalParameterNames": true,
      "defaultEnumHandling": "Integer",
      "flattenInheritanceHierarchy": false,
      "generateKnownTypes": true,
      "generateEnumMappingDescription": false,
      "generateXmlObjects": false,
      "generateAbstractProperties": false,
      "generateAbstractSchemas": true,
      "ignoreObsoleteProperties": false,
      "allowReferencesWithProperties": false,
      "useXmlDocumentation": true,
      "resolveExternalXmlDocumentation": true,
      "excludedTypeNames": [],
      "serviceHost": ".",
      "serviceBasePath": null,
      "serviceSchemes": [],
      "infoTitle": "My Title",
      "infoDescription": null,
      "infoVersion": "1.0.0",
      "documentTemplate": null,
      "documentProcessorTypes": [],
      "operationProcessorTypes": [],
      "typeNameGeneratorType": null,
      "schemaNameGeneratorType": null,
      "contractResolverType": null,
      "serializerSettingsType": null,
      "useDocumentProvider": true,
      "documentName": "v1",
      "aspNetCoreEnvironment": null,
      "createWebHostBuilderMethod": null,
      "startupType": null,
      "allowNullableBodyParameters": true,
      "useHttpAttributeNameAsOperationId": false,
      "output": null,
      "outputType": "Swagger2",
      "newLineBehavior": "Auto",
      "assemblyPaths": [],
      "assemblyConfig": null,
      "referencePaths": [],
      "useNuGetCache": false
    }
  },
  "codeGenerators": {
    "openApiToCSharpClient": {
      "clientBaseClass": "AbstractNSwagClient",
      "configurationClass": "INSwagClientDependencies",
      "generateClientClasses": true,
      "generateClientInterfaces": true,
      "clientBaseInterface": "INSwagClient",
      "injectHttpClient": true,
      "disposeHttpClient": true,
      "protectedMethods": [],
      "generateExceptionClasses": false,
      "exceptionClass": "ApiClientException",
      "wrapDtoExceptions": true,
      "useHttpClientCreationMethod": false,
      "httpClientType": "System.Net.Http.HttpClient",
      "useHttpRequestMessageCreationMethod": false,
      "useBaseUrl": false,
      "generateBaseUrlProperty": false,
      "generateSyncMethods": false,
      "generatePrepareRequestAndProcessResponseAsAsyncMethods": true,
      "exposeJsonSerializerSettings": false,
      "clientClassAccessModifier": "public",
      "typeAccessModifier": "public",
      "generateContractsOutput": false,
      "contractsNamespace": null,
      "contractsOutputFilePath": null,
      "parameterDateTimeFormat": "s",
      "parameterDateFormat": "yyyy-MM-dd",
      "generateUpdateJsonSerializerSettingsMethod": false,
      "useRequestAndResponseSerializationSettings": false,
      "serializeTypeInformation": false,
      "queryNullValue": "",
      "className": "{controller}Client",
      "operationGenerationMode": "MultipleClientsFromOperationId",
      "additionalNamespaceUsages": [
        "CNR.Common.ASP.Collections",
        "CNR.Common.ASP.Exceptions",
        "CNR.Common.ASP.NSwag",
        "CNR.Example.Model",
        "Microsoft.AspNetCore.JsonPatch.Operations"
      ],
      "additionalContractNamespaceUsages": [],
      "generateOptionalParameters": true,
      "generateJsonMethods": false,
      "enforceFlagEnums": false,
      "parameterArrayType": "System.Collections.Generic.IEnumerable",
      "parameterDictionaryType": "System.Collections.Generic.IDictionary",
      "responseArrayType": "System.Collections.Generic.ICollection",
      "responseDictionaryType": "System.Collections.Generic.IDictionary",
      "wrapResponses": false,
      "wrapResponseMethods": [],
      "generateResponseClasses": false,
      "responseClass": "SwaggerResponse",
      "namespace": "CNR.Example.Client",
      "requiredPropertiesMustBeDefined": true,
      "dateType": "System.DateTimeOffset",
      "jsonConverters": null,
      "anyType": "object",
      "dateTimeType": "System.DateTimeOffset",
      "timeType": "System.TimeSpan",
      "timeSpanType": "System.TimeSpan",
      "arrayType": "System.Collections.Generic.ICollection",
      "arrayInstanceType": "System.Collections.ObjectModel.Collection",
      "dictionaryType": "System.Collections.Generic.IDictionary",
      "dictionaryInstanceType": "System.Collections.Generic.Dictionary",
      "arrayBaseType": "System.Collections.ObjectModel.Collection",
      "dictionaryBaseType": "System.Collections.Generic.Dictionary",
      "classStyle": "Poco",
      "jsonLibrary": "NewtonsoftJson",
      "generateDefaultValues": true,
      "generateDataAnnotations": true,
      "excludedTypeNames": [],
      "excludedParameterNames": [ "X-Correlation-ID" ],
      "handleReferences": false,
      "generateImmutableArrayProperties": false,
      "generateImmutableDictionaryProperties": false,
      "jsonSerializerSettingsTransformationMethod": null,
      "inlineNamedArrays": false,
      "inlineNamedDictionaries": false,
      "inlineNamedTuples": true,
      "inlineNamedAny": false,
      "generateDtoTypes": false,
      "generateOptionalPropertiesAsNullable": false,
      "generateNullableReferenceTypes": true,
      "templateDirectory": null,
      "typeNameGeneratorType": null,
      "propertyNameGeneratorType": null,
      "enumNameGeneratorType": null,
      "serviceHost": ".",
      "serviceSchemes": null,
      "output": "../CNR.Example.Client/ExampleClient.Generated.cs",
      "newLineBehavior": "LF"
    }
  }
}
```

**Expected behavior**

C# client is successfully regenerated with minimal changes

**Additional context**

When we migrated from .Net 7 to .Net 8 we had to make some changes to our nswag.json configuration https://github.com/RicoSuter/NSwag/issues/4524#issuecomment-1814801278

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,6 +1,6 @@
 <Project>
   <PropertyGroup>
-    <VersionPrefix>14.3.0</VersionPrefix>
+    <VersionPrefix>14.3.1</VersionPrefix>
 
     <Authors>Rico Suter</Authors>
     <Copyright>Copyright © Rico Suter, 2023</Copyright>

--- a/src/NSwag.Commands/Commands/Generation/AspNetCore/ProjectMetadata.cs
+++ b/src/NSwag.Commands/Commands/Generation/AspNetCore/ProjectMetadata.cs
@@ -83,7 +83,11 @@ namespace NSwag.Commands.Generation.AspNetCore
 
             var args = CreateMsBuildArguments(file, framework, configuration, runtime, noBuild, outputPath);
 
-            var metadata = await TryReadingUsingGetProperties(args, file, noBuild) ?? await ReadUsingMsBuildTargets(args, file, buildExtensionsDir, console);
+            var metadata = await TryReadingUsingGetProperties(args, file, noBuild);
+            if (metadata == null || string.IsNullOrWhiteSpace(metadata[nameof(ProjectDir)]))
+            {
+                metadata = await ReadUsingMsBuildTargets(args, file, buildExtensionsDir, console);
+            }
 
             var platformTarget = metadata[nameof(PlatformTarget)];
             if (platformTarget.Length == 0)
@@ -225,7 +229,8 @@ namespace NSwag.Commands.Generation.AspNetCore
                 nameof(ProjectDir),
                 nameof(ProjectRuntimeConfigFilePath),
                 nameof(TargetFileName),
-                nameof(TargetFrameworkIdentifier)
+                nameof(TargetFrameworkIdentifier),
+                "MSBuildProjectDirectory",
             };
 
             try
@@ -250,6 +255,12 @@ namespace NSwag.Commands.Generation.AspNetCore
                     var metadata = document.RootElement.GetProperty("Properties")
                         .EnumerateObject()
                         .ToDictionary(x => x.Name, x => x.Value.ToString().Trim());
+
+                    // depending on the project type or MSBuild evaluation process, the ProjectDir property may not be set
+                    if (string.IsNullOrWhiteSpace(metadata[nameof(ProjectDir)]))
+                    {
+                        metadata[nameof(ProjectDir)] = metadata["MSBuildProjectDirectory"];
+                    }
 
                     return metadata;
                 }

